src/create-arrow-function.test.ts

```diff
@@ -0,0 +1,192 @@
+import { createArrowFunction } from "./create-arrow-function";
+import { Selection, Position } from "./editor";
+import {
+  createShouldUpdateCodeFor,
+  createShouldNotUpdateCodeFor
+} from "./test-helpers";
+
+const shouldUpdateCodeFor = createShouldUpdateCodeFor(createArrowFunction);
+const shouldNotUpdateCodeFor = createShouldNotUpdateCodeFor(
+  createArrowFunction
+);
+
+describe("Create arrow function declaration from a call expression", () => {
+  it("with nothing else", () => {
+    shouldUpdateCodeFor({
+      code: "readCode();",
+      selection: Selection.cursorAt(0, 0),
+      expectedSnippet: {
+        code: `
+const readCode = () => {
+  \${0:// Implement}
+}`,
+
+        position: new Position(1, 0),
```

Comment:
Actually the position/snippet will generate code that won't work because it puts the arrow function after it's being used.

It works for regular function declarations, but not for an arrow function assigned to a variable:

```ts
readCode();

// Not valid, it should be declared before usage.
const readCode = () => {}
```

Do you need help updating all the test expectations?

Reply:
Hi! I'm sorry, I haven't seen the emails from github notifying my about this.

If you can fix the test expectations that would be great, I'm pretty swamped as of right now.